lib/components/viewers/nearby/stop.tsx

```diff
@@ -105,11 +106,12 @@ const Stop = ({
     new Set()
   )
 
-  // TODO: We need to bring back the day break-up we had with the old stop viewer
   const patternRows = stopData.stoptimesForPatterns
     ?.reduce<PatternStopTime[]>((acc, cur) => {
       const dupe = acc.findIndex(
-        (p) => p.pattern.headsign === cur.pattern.headsign
+        (p) =>
+          extractHeadsignFromPattern(p.pattern) ===
+          extractHeadsignFromPattern(cur.pattern)
```

Comment:
Maybe move `extractHeadsignFromPattern(cur.pattern)` to before line 111.

Reply:
Great clean up thanks!